paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/AddPaymentMethodInteractor.kt

```diff
@@ -0,0 +1,234 @@
+package com.stripe.android.paymentsheet.ui
+
+import com.stripe.android.link.LinkConfigurationCoordinator
+import com.stripe.android.link.ui.inline.InlineSignupViewState
+import com.stripe.android.link.ui.inline.LinkSignupMode
+import com.stripe.android.lpmfoundations.luxe.SupportedPaymentMethod
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.model.PaymentMethodCode
+import com.stripe.android.payments.core.analytics.ErrorReporter
+import com.stripe.android.paymentsheet.forms.FormFieldValues
+import com.stripe.android.paymentsheet.model.PaymentSelection
+import com.stripe.android.paymentsheet.paymentdatacollection.FormArguments
+import com.stripe.android.paymentsheet.paymentdatacollection.ach.USBankAccountFormArguments
+import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel
+import com.stripe.android.uicore.elements.FormElement
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+
+internal interface AddPaymentMethodInteractor {
+    val state: StateFlow<State>
+
+    fun handleViewAction(viewAction: ViewAction)
+
+    fun close()
+
+    data class State(
+        val selectedPaymentMethodCode: PaymentMethodCode,
+        val supportedPaymentMethods: List<SupportedPaymentMethod>,
+        val arguments: FormArguments,
+        val formElements: List<FormElement>,
+        val paymentSelection: PaymentSelection?,
+        val linkSignupMode: LinkSignupMode?,
+        val linkInlineSignupMode: LinkSignupMode?,
+        val processing: Boolean,
+        val usBankAccountFormArguments: USBankAccountFormArguments,
+        val linkConfigurationCoordinator: LinkConfigurationCoordinator,
+    )
+
+    sealed class ViewAction {
+        data class OnPaymentMethodSelected(val code: PaymentMethodCode) : ViewAction()
+        data class OnLinkSignUpStateUpdated(val state: InlineSignupViewState) : ViewAction()
+        data class OnFormFieldValuesChanged(
+            val formValues: FormFieldValues?,
+            val selectedPaymentMethodCode: String
```

Comment:
Not a big deal either way, but this could be a `PaymentMethodCode` rather than `String`. 

We're pretty inconsistent about which we use, and I'm not sure we get much benefit out of it right now.